🧪🚀 CI/CD Pipeline with GitHub Actions.

Rachiid007 opened the pull request:
## 📝 Description

This PR adds a CI/CD pipeline to the project using GitHub actions.
On a push or pull request on the `main` branch, the pipeline will perform the following steps:

- Installing dependencies
- Run tests for the frontend and backend
- If the tests pass, build the frontend otherwise error!
- Deploying the frontend to GitHub pages

## Other changes :wrench:

I deleted the `testing.yml` file as it is no longer needed and added the `cicd.yml` file to run the tests and if they pass, do the frontend build and if it passes also deploy the application to the GitHub pages and otherwise error! (I used the old `testing.yml` file as inspiration to create the new `cicd.yml` file).

I updated the `README.md` file to explain how a developer can configure the project to start working.

I also updated the `vite.config.ts` file and added a new base url for the application. Since GitHub pages will be used to deploy the application, the base url must be the name of the repository for this to work.


![image](https://user-images.githubusercontent.com/91373823/209127359-b2020612-aec1-4d70-935d-263ee0cbcdb4.png)

Rachiid007 committed:
🧪🚀 CI/CD Pipeline with GitHub Actions. #131

remove testing.yml because it is not needed anymore and add cicd.yml to run the tests and if they pass, deploy the app to GitHub Pages.
I updated the README.md file to explain how anyone can use this repo.
I also updated the vite.config.ts file and I add new base url for the app. Because GitHub Pages will be used to deploy the app, the base url will be the repo name.

Pourbaix reviewed (changes requested):
Good job Rachid!

Great implementation of CI/CD System, however there is a little lack of commentaries in the .yml file. (I know you explained it to me but for the other guys in this project, it is better 😉).
Moreover the structure is great so with some commentaries it will be one the best files out there 😄.

In the future, we can even think about a CI/CD implementation for the VPS infrastructure so you can start to think about that too.  

Great PR!

CardinPatson reviewed (approved):
## Good implementation of ci /cd ✨

- Just need more comments in the yml file 
- Need to check tests in the next *PR*
*Reminder the the deploiement of frontend part will be on docker container*

Rachiid007 committed:
🔥 comment(build part)

When we run `npm run build` in /WEB folder we get TypeScript errors.
at the moment we only want to check if all the tests pass!
After that we can add the build part again.
